lightning/src/ln/channel.rs

```diff
@@ -7012,11 +7021,76 @@ impl<SP: Deref> FundedChannel<SP> where
 				log_debug!(logger, "Reconnected channel {} with no loss", &self.context.channel_id());
 			}
 
+			// if next_funding_txid is set:
+			let (commitment_update, tx_signatures, tx_abort) = if let Some(next_funding_txid) = msg.next_funding_txid {
+				if let Some(session) = &self.interactive_tx_signing_session {
+					// if next_funding_txid matches the latest interactive funding transaction:
+					if session.unsigned_tx().compute_txid() == next_funding_txid {
+						debug_assert_eq!(session.unsigned_tx().compute_txid(), self.maybe_get_next_funding_txid().unwrap());
+
+						let commitment_update = if !session.counterparty_sent_tx_signatures() && msg.next_local_commitment_number == 0 {
+							// if it has not received tx_signatures for that funding transaction AND
+							// if next_commitment_number is zero:
+							//   MUST retransmit its commitment_signed for that funding transaction.
+							let commitment_signed = self.context.get_initial_commitment_signed(&self.funding, logger)?;
+							Some(msgs::CommitmentUpdate {
+								commitment_signed,
+								update_add_htlcs: vec![],
+								update_fulfill_htlcs: vec![],
+								update_fail_htlcs: vec![],
+								update_fail_malformed_htlcs: vec![],
+								update_fee: None,
+							})
+						} else { None };
+						// if it has not received tx_signatures for that funding transaction AND
+						// if it has already received commitment_signed AND it should sign first, as specified in the tx_signatures requirements:
+						//   MUST send its tx_signatures for that funding transaction.
+						// else if it HAS received commitment_signed AND has received tx_signatures for that funding transaction:
+						//   MUST send its tx_signatures for that funding transaction.
+						let tx_signatures = if session.has_received_commitment_signed() && ((
+							!session.counterparty_sent_tx_signatures() &&
+							session.holder_sends_tx_signatures_first()
+						) || session.counterparty_sent_tx_signatures()) {
```

Comment:
Shouldn't the last closing parenthesis  be before the `||`? Or rather shouldn't it be removed along with it's opening pair? The spec doesn't contingent `session.counterparty_sent_tx_signatures()` on `session.has_received_commitment_signed()`, if I'm reading it correctly.

Reply:
Oh yip. That extra constraint slipped in, possibly when I updated this section after Matt opened PR 1214 on the spec.